bukuserver/response.py

```diff
@@ -1,4 +1,45 @@
-response_template = {
-    "success": {'status': 0, 'message': 'success'},
-    "failure": {'status': 1, 'message': 'failure'}
-}
+from enum import Enum
+from flask import jsonify
+from flask_api import status
+
+OK, FAIL = 0, 1
+
+class Response(Enum):
+    SUCCESS = (status.HTTP_200_OK, "Success.")
+    FAILURE = (status.HTTP_400_BAD_REQUEST, "Failure.")
+    INPUT_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Request not valid")
+    BOOKMARK_NOT_FOUND = (status.HTTP_404_NOT_FOUND, "Bookmark not found.")
+    TAGS_MISSING = (status.HTTP_400_BAD_REQUEST, "Replacement tags are missing.")
+    TAG_NOT_FOUND = (status.HTTP_404_NOT_FOUND, "Tag not found.")
+    TAGS_NOT_LIST = (status.HTTP_400_BAD_REQUEST, "List of tags expected.")
+    TAG_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Invalid tag.")
+
+    @staticmethod
+    def from_flag(flag: bool):
+        return Response.SUCCESS() if flag else Response.FAILURE()
+
+    @property
+    def status_code(self):
+        return self.value[0]
+
+    @property
+    def message(self):
+        return self.value[1]
+
+    @property
+    def status(self):
+        return OK if self.status_code == status.HTTP_200_OK else FAIL
+
+    def get_json(self, data :dict = None):
+        json = {'status': self.status, 'message': self.message}
+        if isinstance(data, dict):
+            json.update(data)
+        return json
+
+    def __call__(self, *args, **kwargs):
+        """
+        Kwargs
+           data (dict): updates the response's JSON
+        """
```

Comment:
…If your goal was to have a single optional argument that must be explicitly named in the invocation, this might be a better way of doing it:
```py
def __call__ (self, *, data :dict = None):
```